sites/partners/src/components/listings/ListingFormActions.tsx

```diff
@@ -195,60 +200,249 @@ const ListingFormActions = ({
+      if (type === "edit") {
+        if (isListingApprover) {
+          // admins can publish a draft
+          if (listing.status === ListingStatus.pending) elements.push(publishButton)
+          // admins can approve and publish a pending approval or changes requested listing
+          if (
+            listing.status === ListingStatus.pendingReview ||
+            listing.status === ListingStatus.changesRequested
+          )
+            elements.push(approveAndPublishButton)
+          // admins can reopen a closed listing
+          if (listing.status === ListingStatus.closed) elements.push(reopenButton)
+        } else {
+          // partners can submit for approval a draft or changes requested listing
+          if (
+            listing.status === ListingStatus.pending ||
+            listing.status === ListingStatus.changesRequested
+          )
+            elements.push(submitButton)
+        }
+
+        // all users can make updates to an open listing
+        elements.push(saveExitButton)
+
+        // admins can request changes on pending review listings
+        if (isListingApprover && listing.status === ListingStatus.pendingReview)
+          elements.push(requestChangesButton)
+
+        // all users can unpublish a closed listing
+        if (listing.status === ListingStatus.closed) {
+          elements.push(unpublishButton)
+        }
+
+        // all users can close or unpublish open listings
+        if (listing.status === ListingStatus.active) {
+          elements.push(closeButton)
+          elements.push(unpublishButton)
+        }
+
+        const lotteryResults = listing?.events?.find(
+          (event) => event.type === ListingEventType.lotteryResults
         )
-        elements.push(viewPostedResultsButton(eventUrl))
+
+        // all users can manage lottery results on closed listings
+        if (lotteryResults) {
+          elements.push(editPostedResultsButton(lotteryResults))
+        } else if (listing.status === ListingStatus.closed) {
+          elements.push(postResultsButton)
+        }
+
+        elements.push(cancelButton)
       }
+      return elements
     }
 
-    // new unsaved listing
-    if (type === "add") {
-      elements.push(publishButton)
-      elements.push(saveDraftButton)
-      elements.push(cancelButton)
-    }
+    const getCoreActions = () => {
+      const elements = []
+      // read-only form
+      if (type === "details") {
+        elements.push(editFromDetailButton)
+        elements.push(previewButton)
 
-    // listing saved at least once
-    if (type === "edit") {
-      elements.push(saveExitButton)
+        if (listing.events.find((event) => event.type === ListingEventType.lotteryResults)) {
+          const eventUrl = pdfUrlFromListingEvents(
+            listing?.events,
+            ListingEventType.lotteryResults,
+            process.env.cloudinaryCloudName
+          )
+          elements.push(viewPostedResultsButton(eventUrl))
+        }
```

Comment:
This section is repeated in the approval flow, could it be pulled out so it's reusable?